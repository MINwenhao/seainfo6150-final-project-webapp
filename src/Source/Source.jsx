import React from 'react';
import styles from "./Source.module.css";


const Source = (prop) => {
    let url = '/detail/' + (prop.source.uuid)
    return (
        <div >
            <a href={url}>
                <img src={prop.source.pictures[0]}
                    alt="first pic of the source" className={styles.img} /></a>
            <p>{prop.source.location}</p>
        </div>

    )

}

export default Source
