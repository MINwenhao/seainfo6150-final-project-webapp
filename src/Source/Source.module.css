.img {
    width: 250px;
    height: 120px;

}

.source {
    grid-column: 1;
}