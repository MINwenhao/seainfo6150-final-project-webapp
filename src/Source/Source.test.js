import { render } from "@testing-library/react";
import React from "react";
import Source from "./Source.jsx";
import {BrowserRouter} from "react-router-dom";

describe("AddingMachineHooks tests", () => {
    const source={
        "uuid": "0005",
        "city":"Seattle",
        "location": "401 Terry Ave N",
        "category": "house",
        "type":"3 Bed",
        "rentTime":"long-term",
        "owner": {
            "type":"real estate",
          "portrait": "http://clipart-library.com/images/gieEdLqzT.jpg",
          "name":"NEU",
          "connect":"real-eatate@neu.com",
          "rate":5
        },
        "updateDate": "November 22nd 2019, 7:12 am",
        "pictures":["https://cdn.pixabay.com/photo/2014/08/11/21/32/fireplace-416042_1280.jpg",
        "https://cdn.pixabay.com/photo/2015/09/08/22/03/luggage-930804_1280.jpg",
        "https://cdn.pixabay.com/photo/2017/02/24/12/18/apartment-2094654_1280.jpg"],
        "price":"5000",
        "unit":"month",
        "rate":5,
        "recommend":5,
        "available":true,
        "CommunityAmenities":["Walk to Amazon Campus","Controlled Access Garage","Easy Access to Highways and Transit","Resident Social Events"],
        "ApartmentAmenities":["Spacious Closets","Private Balcony","Quartz Countertops","Vaulted Ceilings"]
      }
      describe('Test suits for MyComponentWithLink', () => {
        it('should match with snapshot', () => {
        const tree = render(
            <BrowserRouter>
              <Source source={source} key={source.uuid} />
            </BrowserRouter>
          )
          
         expect(tree).toMatchSnapshot();
         });
    });
});