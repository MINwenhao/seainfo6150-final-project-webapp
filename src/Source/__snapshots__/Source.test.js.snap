// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddingMachineHooks tests Test suits for MyComponentWithLink should match with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <a
          href="/detail/0005"
        >
          <img
            alt="first pic of the source"
            class="img"
            src="https://cdn.pixabay.com/photo/2014/08/11/21/32/fireplace-416042_1280.jpg"
          />
        </a>
        <p>
          401 Terry Ave N
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <a
        href="/detail/0005"
      >
        <img
          alt="first pic of the source"
          class="img"
          src="https://cdn.pixabay.com/photo/2014/08/11/21/32/fireplace-416042_1280.jpg"
        />
      </a>
      <p>
        401 Terry Ave N
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
